README.md

```diff
@@ -1,2 +1,112 @@
 # ilib-mono
-Monorepo for containing the plethora of ilib tools and plugins
+
+Monorepo for containing the plethora of ilib tools and plugins.
+
+## Setting up Circleci for Your Package
+
+To set up Circleci for your package, you need to do the following:
+
+1. Edit the .circleci/config.yml file to include your package in the
+   list of packages to build and test. Modify the list of packages in
+   the workflows.setup-workflow.jobs[0].modules section to include your
+   package. This will cause Circleci to build and test only your package
+   whenever a new commit is pushed to the repository in your package.
+
+2. Include the .circleci/config.yml from the original repo in your
+   package. This needs to be modified a little from the original to
+   differentiate between the workflows and tests for each package. The
+   config.yml files for all packages are merged together to create the
+   final config.yml file for Circleci, so if your workflow is named the
+   same thing as another package's workflow, it will overwrite the other
+   package's workflow. Additionally, we have to cd to the package directory
+   before running the tests. This is because the original config.yml file
+   assumes that the package is in the root of the repository. Of course,
+   this is not the case for a monorepo!
+
+   Example original .circleci/config.yml file:
+
+   ```yaml
+version: 2.1
+jobs:
+  test:
+    parameters:
+        docker_image:
+          type: string
+          default: cimg/node:current-browsers
+    docker:
+      - image: << parameters.docker_image >>
+    steps:
+      - checkout
+      - run:
+          name: Setup
+          command: |
+            rm -rf node_modules package-lock.json
+            npm install
```

Comment:
Is there a reason for using `npm` instead of `yarn`?
Shall we use `yarn` instead?
Since we're creating brand new repo, I'd love to stick to modern stack whenever possible.

Reply:
Few quotes from `yarn` creators:

> _The npm client installs dependencies into the `node_modules` directory non-deterministically. This means that based on the order dependencies are installed, the structure of a `node_modules` directory could be different from one person to another. These differences can cause “works on my machine” bugs_ _that take a long time to hunt down._

> Yarn resolves these issues around versioning and non-determinism by using lockfiles and an install algorithm that is deterministic and reliable. 

> (...) Yarn is able to parallelize operations, which maximizes resource utilization and makes the install process faster.

Source: [Yarn: A new package manager for JavaScript ](https://engineering.fb.com/2016/10/11/web/yarn-a-new-package-manager-for-javascript/)